libs/core/threading_base/src/annotated_function.cpp

```diff
@@ -0,0 +1,23 @@
+//  Copyright (c) 2021 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/config.hpp>
+#if defined(HPX_HAVE_THREAD_DESCRIPTION) && defined(HPX_HAVE_APEX)
+#include <hpx/threading_base/annotated_function.hpp>
+
+#include <string>
+#include <unordered_set>
+#include <utility>
+
+namespace hpx { namespace util { namespace detail {
+    char const* store_function_annotation(std::string name)
```

Comment:
Minor nitpick: if this function is always used with an rvalue (and seems it is), you might want to have it accept a `std::string&& name` instead. This is not a big deal however - it saves just one move operation.

Reply:
Another fair point. If we're going to do extra work to make the strings live longer, might as well do it as efficiently as possible. It's an internal API anyway so we can always change it later if there's another use case. I'll change that as well.